modelbuilders_bench/catboost_mb.py

```diff
@@ -0,0 +1,211 @@
+                    help='Count Pool creation in time measurements')
+
+parser.add_argument('--grow-policy', type=str, default='Depthwise',
+                    help='Controls a way new nodes are added to the tree')
+
+parser.add_argument('--learning-rate', '--eta', type=float, default=0.3,
+                    help='Step size shrinkage used in update '
+                    'to prevents overfitting')
+
+parser.add_argument('--max-bin', type=int, default=256,
+                    help='Maximum number of discrete bins to '
+                    'bucket continuous features')
+
+parser.add_argument('--max-depth', type=int, default=6,
+                    help='Maximum depth of a tree')
+
+parser.add_argument('--max-leaves', type=int, default=0,
+                    help='Maximum number of nodes to be added')
+
+parser.add_argument('--n-estimators', type=int, default=100,
+                    help='Number of gradient boosted trees')
+
+parser.add_argument('--objective', type=str, required=True,
+                    choices=('RMSE', 'Logloss',
+                             'multi:softmax', 'multi:softprob'),
+                    help='Control a balance of positive and negative weights')
+
+parser.add_argument('--reg-lambda', type=float, default=1,
+                    help='L2 regularization term on weights')
+
+parser.add_argument('--scale-pos-weight', type=float, default=1,
+                    help='Controls a balance of positive and negative weights')
+
+parser.add_argument('--subsample', type=float, default=1,
+                    help='Subsample ratio of the training instances')
+
+params = bench.parse_args(parser)
+
+X_train, X_test, y_train, y_test = bench.load_data(params)
+
+cb_params = {
+    'verbose': 0,
+    'learning_rate': params.learning_rate,
+    'max_depth': params.max_depth,
+    'subsample': params.subsample,
+    'colsample_bylevel': 1,
+    'reg_lambda': params.reg_lambda,
+    'grow_policy': params.grow_policy,
+    'max_bin': params.max_bin,
+    'objective': params.objective,
+    'random_seed': params.seed,
+    'iterations': params.n_estimators,
+}
+
+# CatBoost restriction
+if cb_params['grow_policy'] == 'Lossguide':
+    cb_params['max_leaves'] = params.max_leaves
+
+if params.threads != -1:
+    cb_params.update({'thread_count': params.threads})
+
+metric_name: tp.List[str]
+metric_func: tp.List[tp.Callable]
+
+if params.objective == "RMSE":
+    task = 'regression'
+    metric_name = ['rmse', 'r2']
+    metric_func = [bench.rmse_score, bench.r2_score]
+else:
+    task = 'classification'
+    metric_name = ['accuracy', 'log_loss']
+    metric_func = [bench.accuracy_score, bench.log_loss]
+    if 'cudf' in str(type(y_train)):
+        params.n_classes = y_train[y_train.columns[0]].nunique()
+    else:
+        params.n_classes = len(np.unique(y_train))
+
+    # Covtype has one class more than there is in train
+    if params.dataset_name == 'covtype':
+        params.n_classes += 1
```

Comment:
@razdoburdin, this should be aligned to https://github.com/IntelPython/scikit-learn_bench/pull/125